### PR TITLE
[jwplayer] Update `registerPlugin` types

### PR DESCRIPTION
**Pull request by amowu**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.jwplayer.com/players/docs/jw8-add-a-custom-or-third-party-plugin#add-and-register-a-plugin
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/jwplayer/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 092131d5582545555a2ac55fee80f9595d66d9b2
</p>

File: types/jwplayer/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "private": true,
     "name": "@types/jwplayer",
-    "version": "8.28.9999",
+    "version": "8.31.0",
```
Comment (codershiba) [inline types/jwplayer/package.json:4]:
```suggestion
    "version": "8.31.9999",
```

**Review by esetnik — APPROVED** [review]

**Review by MysteryBlokHed — COMMENTED** [review]
I'd recommend adding a test to [`jwplayer-tests.ts`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/jwplayer/jwplayer-tests.ts) to make sure that this change works as intended

**Review by MysteryBlokHed — APPROVED** [review]
LGTM—you can also add a test for `registerPlugin` if you'd like and I'll reapprove

**Comment by amowu** [comment]
> LGTM—you can also add a test for `registerPlugin` if you'd like and I'll reapprove

Thank you for your review, but I see that this method is already written in the tests.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/41aa59f8e323a93b98ecea69fcf67a7a0a2cfa67/types/jwplayer/jwplayer-tests.ts#L218-L219

**Comment by MysteryBlokHed** [comment]
> > LGTM—you can also add a test for `registerPlugin` if you'd like and I'll reapprove
> 
> Thank you for your review, but I see that this method is already written in the tests.
> 
> https://github.com/DefinitelyTyped/DefinitelyTyped/blob/41aa59f8e323a93b98ecea69fcf67a7a0a2cfa67/types/jwplayer/jwplayer-tests.ts#L218-L219

I just meant to test the new properties (i.e. the parameters for the function passed to `jsPlugin`). If you'd like, you can either add the properties to the existing test on line 219, or add a line under it to test the new changes

**Review by nlhiorth — APPROVED** [review]

**Comment by amowu** [comment]
Ready to merge

**Pull request merged**